app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CalendarGridAdapter.java

```diff
@@ -0,0 +1,87 @@
+package com.github.onedirection.navigation.fragment.calendar;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.github.onedirection.R;
+import com.github.onedirection.events.Event;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+public class CalendarGridAdapter extends ArrayAdapter {
+    private Calendar currentDate;
+    private List<Event> events;
+    private LayoutInflater inflater;
+    private final List<Date> dates;
+
+    public CalendarGridAdapter(@NonNull Context context, List<Date> dates, Calendar currentDate, List<Event> events) {
+        super(context, R.layout.single_cell_layout);
+        this.dates = dates;
+        this.currentDate = currentDate;
+        this.events = events;
+        inflater = LayoutInflater.from(context);
```

Comment:
[Detail]
```suggestion
        this.inflater = LayoutInflater.from(context);
```
for consistency.
Also, it's easier to read if the attributes are initialized in the same order as they are declared, but not that we really need this here :)